llvm/labs/lab2/durandin_vladimir/vladimir_inline_pass.cpp

```diff
@@ -0,0 +1,165 @@
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+#include <string>
+
+using namespace llvm;
+
+namespace {
+Instruction *getNextNonTerminator(BasicBlock *BB, Instruction *I) {
+  for (auto &Inst : *BB) {
+    if (&Inst == I) {
+      for (auto &NextInst :
```

Comment:
No, I'm not expecting instructions after the terminator. I'm just trying to get a pointer to the next instruction after `Instruction I`.
If there is such an instruction, then I get a pointer to it. Otherwise I return a `nullptr`.

Reply:
Still, yes, you are right...
It's okay if the next function is a terminator. Fixed this bug.